providers/snapchatads/handlers.go

```diff
@@ -98,3 +101,110 @@ func (c *Connector) parseReadResponse(
 		params.Fields,
 	)
 }
+
+func (c *Connector) buildWriteRequest(ctx context.Context, params common.WriteParams) (*http.Request, error) {
+	url, err := c.constructURL(params.ObjectName)
+	if err != nil {
+		return nil, err
+	}
+
+	method := http.MethodPost
+
+	// When updating an object, the record ID is not included in the URL path; instead,
+	// it must be provided in the body parameters.
+	// EX. Refer https://developers.snap.com/api/marketing-api/Ads-API/billing-centers#update-a-billing-center.
+	if params.RecordId != "" {
```

Comment:
can you make sure that the json body contains the record ID in case this is set?